src/fibex.rs

```diff
@@ -0,0 +1,2274 @@
+// fibex parser
+
+use quick_xml::{
+    events::{BytesStart, Event as XmlEvent},
+    Reader as XmlReader,
+};
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    fs::File,
+    io::{BufRead, BufReader, Read},
+    path::Path,
+    rc::Rc,
+};
+use thiserror::Error;
+
+const ERROR_TAG: &str = "FIBEX Error";
+
+#[derive(Error, Debug)]
+pub enum FibexError {
+    #[error("{}: {0}", ERROR_TAG)]
+    Parse(String),
+    #[error("{}: {0}", ERROR_TAG)]
+    Model(String),
+    #[error("{}: {0:?}", ERROR_TAG)]
+    Xml(#[from] quick_xml::Error),
+}
+
+pub mod meta {
+    use super::*;
+
+    #[derive(Debug)]
+    pub struct FibexModel {
+        pub services: Vec<FibexServiceInterface>,
+        pub types: Vec<FibexTypeReference>,
+    }
+
+    impl FibexModel {
+        pub fn new() -> FibexModel {
+            FibexModel {
+                services: Vec::new(),
+                types: Vec::new(),
+            }
+        }
+
+        pub fn get_request(
+            &self,
+            service_id: usize,
+            service_version: usize,
+            method_id: usize,
+        ) -> Option<&FibexTypeDeclaration> {
+            return self
```

Comment:
in Rust the last statement without a semicolon represents the return value => better not to use `return` when possible

Reply:
done